src/kernels/execution/kernelCompletionPreWarmer.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { inject, injectable } from 'inversify';
+import { IExtensionSyncActivationService } from '../../platform/activation/types';
+import { IDisposableRegistry } from '../../platform/common/types';
+import { isPythonKernelConnection } from '../helpers';
+import { IKernelConnectionSession, IKernelProvider } from '../types';
+
+@injectable()
+export class KernelCompletionsPreWarmer implements IExtensionSyncActivationService {
```

Comment:
Moved out of the kernel.ts file, there was no need for this to be inside kernel.ts